src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -1052,14 +1056,53 @@ julia> project(P, [1, 2]) |> constraints_list
 ```
 """
 function project(P::AbstractPolyhedron{N}, block::AbstractVector{Int}) where {N}
-    if constrained_dimensions(P) ⊆ block
-        clist = [HalfSpace(c.a[block], c.b) for c in constraints_list(P)]
-    else
+    general_case = false
+
+    # cheap case
+    clist = nothing  # allocate later
+    @inbounds for c in constraints(P)
```

Comment:
does the check `constrained_dimensions(P) ⊆ block` still apply? and if it is false => use the general case

Reply:
The old check is part of the new check. `constrained_dimensions(P)` goes through all constraints of `P` to collect the constrained dimensions in *all* constraints. The new check does not require the aggregated constrained dimensions. First doing the old check would essentially do the same work twice and does not simplify the code because we would still need to have the current code again for having the general case.